backend/core/interactem/core/models/canonical.py

```diff
@@ -0,0 +1,64 @@
+from collections.abc import Sequence
+
+from pydantic import BaseModel
+
+from interactem.core.models.base import IdType, NodeType, PortType
+from interactem.core.models.user import (
+    ParallelConfig,
+    UserOperatorID,
+    UserOperatorParameter,
+    UserOperatorTag,
+)
+
+CanonicalOperatorID = UserOperatorID
+CanonicalPortID = IdType
+CanonicalPipelineID = IdType
+CanonicalPipelineRevisionID = int
+
+
+class CanonicalPort(BaseModel):
+    id: CanonicalPortID
+    node_type: NodeType = NodeType.port
+    port_type: PortType
+    operator_id: UserOperatorID  # The operator this port belongs to
+    portkey: str  # Unused currently, but can be used for port-port compatibility
+
+
+class CanonicalInput(CanonicalPort):
+    port_type: PortType = PortType.input
+
+
+class CanonicalOutput(CanonicalPort):
+    port_type: PortType = PortType.output
+
+
+class CanonicalOperator(BaseModel):
```

Comment:
Lets add a doc string describing what this is for future selves, is this the "registered" operator?